corehq/apps/domain/views.py

```diff
@@ -565,6 +568,12 @@ def test_repeater(request, domain):
         fake_post = generator.get_test_payload(domain)
         headers = generator.get_headers()
 
+        if use_basic_auth == 'true':
+            username = request.POST.get('username')
+            password = request.POST.get('password')
+            user_pass = base64.encodestring(':'.join((username, password))).replace('\n', '')
```

Comment:
this looks like it was copied from `Repeater.get_headers`. Maybe you could pull this out into a nicely named util that is used by both methods?